src/mono/wasm/runtime/assets.ts

```diff
@@ -482,31 +561,7 @@ export function mono_wasm_load_data_archive(data: Uint8Array, prefix: string): b
     }
 
     data = data.slice(manifestSize + 8);
-
-    // Create the folder structure
-    // /usr/share/zoneinfo
-    // /usr/share/zoneinfo/Africa
-    // /usr/share/zoneinfo/Asia
-    // ..
-
-    const folders = new Set<string>();
-    manifest.filter(m => {
-        const file = m[0];
-        const last = file.lastIndexOf("/");
-        const directory = file.slice(0, last + 1);
-        folders.add(directory);
-    });
-    folders.forEach(folder => {
-        Module["FS_createPath"](prefix, folder, true, true);
-    });
-
-    for (const row of manifest) {
-        const name = row[0];
-        const length = row[1];
-        const bytes = data.slice(0, length);
-        Module["FS_createDataFile"](prefix, name, bytes, true, true);
```

Comment:
Here we are writing data to the file we created. Why then are we creating nodes in FS in the new approach with read-only-mode?

Reply:
Why would the application try to write into timezone data ? Making it RW would be more code and complexity.